server/api/user_handler.py

```diff
@@ -48,6 +55,35 @@ def login():
     if not bcrypt.check_password_hash(user.password, password):
         return jsonify({"Error": "incorrect password"}), 400
 
-    # TODO: register an auth token and return to the user
+    token = jwt.encode({"user": username, "exp": datetime.utcnow() + timedelta(minutes=exp)}, \
+                       app.app.config['JWT_SECRET'])
+    return jsonify({"auth_token": token}), 201
+
+
+def require_auth(route):
```

Comment:
you probably want to move this into a folder called `middleware` but the function looks good